pkg/controllers/jobset_controller.go

```diff
@@ -255,10 +255,16 @@ func (r *JobSetReconciler) resumeJobSetIfNecessary(ctx context.Context, js *jobs
 				log.Error(nil, "job missing ReplicatedJobName label")
 			}
 			job.Spec.Suspend = pointer.Bool(false)
-			job.Status.StartTime = nil
 			if err := r.Update(ctx, job); err != nil {
 				return err
 			}
+			if job.Status.StartTime == nil {
+				continue
+			}
+			job.Status.StartTime = nil
+			if err := r.Status().Update(ctx, job); err != nil {
+				return err
+			}
```

Comment:
this is less intuitive than simply encapsulate the logic under the condition:
```suggestion
			if job.Status.StartTime != nil {
			  job.Status.StartTime = nil
			  if err := r.Status().Update(ctx, job); err != nil {
			 	return err
			  }
			}
```